packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -911,17 +912,20 @@ function mapToNativeAppSyncAuthDirectives(
       };
       rule.groups.forEach((group) => groupProvider.get(provider)?.add(group));
     } else {
+      //generalProviderUsed.add(provider);
       rules.add(provider);
     }
   }
 
   groupProvider.forEach((groups, provider) => {
-    rules.add(
-      `${provider}(cognito_groups: [${Array.from(groups)
-        .map((group) => `"${group}"`)
-        .join(', ')}])`,
-    );
-    // example: (cognito_groups: ["Bloggers", "Readers"])
+    if(!generalProviderUsed.has(provider)) {
+      rules.add(
+        `${provider}(cognito_groups: [${Array.from(groups)
+          .map((group) => `"${group}"`)
+          .join(', ')}])`,
```

Comment:
Can replace `map` and `join` with a single `reduce`. Also might be worth extracting a util here, e.g.
```ts
// contrived name
const getGroupsValue = (groups) => Array.from(groups)
          .map((group) => `"${group}"`)
          .join(', ')
```

Reply:
I can and have replaced this with reduce.

I'll own that I prefer the readability of map | join and it implicitly solves the interior comma problem that I need to solve explicitly with reduce. Not a big enough issue that its worth advocating for one vs the other.